### PR TITLE
Refactored code in TableServiceImpl to remove the cyclic dependency between TableServiceImpl and MViewServiceImpl.

### PR DESCRIPTION
**Pull request by ajoymajumdar**
1. Delete the views of a table when a table gets deleted.
2. Rename the views of a table when a table gets renamed.

### CODE REVIEW

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatDeleteTablePostEvent.java [hunk 1]
```diff
@@ -51,7 +52,27 @@ public MetacatDeleteTablePostEvent(
...
+     * Constructor.
+     *
+     * @param name           name
+     * @param requestContext context
+     * @param source         The source object which threw this event
+     * @param table          table info
+     * @param isMView        true, if the table is a materialized view
+     */
+    public MetacatDeleteTablePostEvent(
+        @Nonnull @NonNull final QualifiedName name,
```
Comment (tgianos) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatDeleteTablePostEvent.java]: Didn't we decide not to put these annotations in?

File: metacat-main/src/main/java/com/netflix/metacat/main/services/MViewServiceEventHandler.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    private final UserMetadataService userMetadataService;
+
+    /**
+     * Default constructor.
+     * @param config               server configurations
+     * @param mViewService         view service
+     * @param userMetadataService  user metadata service
+     */
+    @Autowired
+    public MViewServiceEventHandler(final Config config, final MViewService mViewService,
```
Comment (tgianos) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/MViewServiceEventHandler.java]: you're killing me :-)

File: metacat-main/src/main/java/com/netflix/metacat/main/services/MViewServiceEventHandler.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+     *
+     * @param event event
+     */
+    @EventListener
+    public void metacatDeleteTablePostEventHandler(final MetacatDeleteTablePostEvent event) {
+        if (config.canCascadeViewsMetadataOnTableDelete() && !event.isMView()) {
+            final QualifiedName name = event.getTable().getName();
+            try {
+                // delete views associated with this table
+                final List<NameDateDto> viewNames = mViewService.list(name);
```
Comment (tgianos) [inline metacat-main/src/main/java/com/netflix/metacat/main/services/MViewServiceEventHandler.java]: any reason to even bother saving this to a local variable and not just doing the whole thing in one statement?

File: metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatDeleteTablePostEvent.java (hunk 1 above)
Comment (ajoymajumdar) [inline metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatDeleteTablePostEvent.java]: I do not see it being removed from all classes yet.

**Pull request merged**